api/src/pcapi/core/providers/titelive_apply_new_filters_on_history.py

```diff
@@ -0,0 +1,348 @@
+    """
+    Cancel multiple bookings and update the users' credit information on Batch.
+    Note that this will not reindex the stock.offer in Algolia
+    """
+    deleted_bookings: list[bookings_models.Booking] = []
+    for booking in stock.bookings:
+        if not dry and bookings_api._cancel_booking(
+            booking, reason, cancel_even_if_used=typing.cast(bool, stock.offer.isEvent)
+        ):
+            deleted_bookings.append(booking)
+        else:
+            deleted_bookings.append(
+                booking
+            )  # This doesn't exclude certain booking while in test mode because we don't handle exceptions
+    return deleted_bookings
+
+
+# This if the exact copy of bookings_api.cancel_bookings_from_rejected_offer with dry run support
+def cancel_bookings_from_rejected_offer(offer: models.Offer, dry: bool = False) -> list[bookings_models.Booking]:
+    cancelled_bookings = []
+    for stock in offer.stocks:
+        cancelled_bookings.extend(_cancel_bookings_from_stock(stock, BookingCancellationReasons.FRAUD, dry=dry))
+    logger.info(
+        "Cancelled bookings for rejected offer",
+        extra={
+            "bookings": [b.id for b in cancelled_bookings],
+            "offer": offer.id,
+        },
+    )
+
+    return cancelled_bookings
+
+
+# This if the exact copy of offers_api.reject_inappropriate_products with dry run support and return list of email instead of bool
+def reject_inappropriate_products(
+    ean: str, send_booking_cancellation_emails: bool = True, dry: bool = False
+) -> list[str]:
+    products = models.Product.query.filter(models.Product.extraData["ean"].astext == ean).all()
+    emails: list[str] = []
+
+    if not products:
+        return emails
+
+    for product in products:
+        product.isGcuCompatible = False
+        db.session.add(product)
+
+    offers_query = models.Offer.query.filter(
+        models.Offer.productId.in_(p.id for p in products),
+        models.Offer.validation != models.OfferValidationStatus.REJECTED,
+    ).options(sa.orm.joinedload(models.Offer.stocks).joinedload(models.Stock.bookings))
+
+    offers = offers_query.all()
+
+    if not dry:
+        offers_query.update(
+            values={
+                "validation": models.OfferValidationStatus.REJECTED,
+                "lastValidationDate": datetime.datetime.utcnow(),
+                "lastValidationType": OfferValidationType.CGU_INCOMPATIBLE_PRODUCT,
+            },
+            synchronize_session="fetch",
+        )
+
+    offer_ids = []
+    for offer in offers:
+        offer_ids.append(offer.id)
+        bookings = cancel_bookings_from_rejected_offer(offer, dry=dry)
+        for booking in bookings:
+            emails.append(booking.user.email)
+
+        if send_booking_cancellation_emails and not dry:
+            for booking in bookings:
+                send_booking_cancellation_emails_to_user_and_offerer(booking, reason=BookingCancellationReasons.FRAUD)
```

Comment:
Puisqu'on ne veut pas appeler `send_booking_cancellation_emails_to_user_and_offerer()` dans ce script (car ça envoie un mail au pro et pas au bénéficiaire, alors qu'on veut faire l'inverse), autant commenter ce bloc de code. C'est plus clair pour tout le monde : 

```suggestion
        # `send_booking_cancellation_emails_to_user_and_offerer` would send an e-mail to the pro and
        # none to the beneficiary. We want the opposite. E-mails will be sent by a separate script.
        # if send_booking_cancellation_emails:
        #     for booking in bookings:
        #         send_booking_cancellation_emails_to_user_and_offerer(booking, reason=BookingCancellationReasons.FRAUD)
```

Reply:
Oui agree.